R/generate_js.R

```diff
@@ -17,9 +17,15 @@ generate_js <- function(ir, options) {
   eqs <- generate_js_equations(dat, rewrite)
   core <- generate_js_core(eqs, dat, rewrite)
 
+  fns <- unlist(lapply(dat$equations, function(x) x$depends$functions),
+                FALSE, FALSE)
+  uses_sum <- "sum" %in% fns
+
   ## This is all we need to dump out
   list(code = generate_js_generator(core, dat),
-       name = dat$config$base)
+       name = dat$config$base,
+       include = c(interpolate.js = dat$features$has_interpolate,
+                   support_sum.js = uses_sum))
```

Comment:
these changes break the [less basic model](https://github.com/mrc-ide/squire/blob/9bd90684db35267b7f408ca90a77990af9549701/inst/odin/less_basic_model_for_js.R)

```
ReferenceError: odinSum2 is not defined
    at less_basic_model_for_js.require.odin.less_basic_model_for_js.less_basic_model_for_js.rhs (/opt/build/squire.js:1077:32)
    at Dopri5.rhs (/opt/build/squire.js:758:13)
    at Dopri5.reset (/opt/build/squire.js:478:14)
    at Dopri.initialise (/opt/build/squire.js:146:23)
    at Object.integrate (/opt/build/squire.js:116:12)
    at Object.integrate (/opt/build/squire.js:559:24)
    at integrateOdin (/opt/build/squire.js:771:21)
    at less_basic_model_for_js.require.odin.less_basic_model_for_js.less_basic_model_for_js.run (/opt/build/squire.js:1119:12)
    at runModel (/opt/build/squire.js:1189:14)
    at Object.<anonymous> (/opt/e2e/test.js:3:15)
```